include/boost/geometry/algorithms/detail/overlay/get_turn_info_la.hpp

```diff
@@ -736,20 +737,12 @@ struct get_turn_info_linear_areal
 
                 if ( ip0.is_qj )
                 {
-                    // TODO: wrong! q should never be substituted for p.
-                    // In the end point there is no point "k", that should be fixed in another way
-                    ov::side_calculator_for_endpoint
-                        <
-                            typename IntersectionInfo::cs_tag,
-                            rp1_type, rp2_type,
-                            typename IntersectionInfo::side_strategy_type,
-                            rp2_type
-                        > side_calc(inters.rqi(), inters.rpi(), inters.rpj(),
-                                    inters.rqi(), inters.rqj(), inters.rqk(),
-                                    inters.get_side_strategy());
+                    int const side_pj_q2 = sides.apply(range_q.at(1), range_q.at(2), range_p.at(1));
+                    int const side_pj_x = sides.apply(range_q.at(0), range_p.at(0), range_p.at(1));
+                    int const side_qk_x = sides.apply(range_q.at(0), range_p.at(0), range_q.at(2));
```

Comment:
Is this new way of expressing points more readable to you (e.g. instead of `i()`, `j()`, `k()`)? For me I don't know, looks confusing. Granted, if someone doesn't know what `i`, `j` or  `k` means it's also confusing but at least one cane distinguish one from the other on first sight. Maybe it's a matter of getting used to it. Is it? How do you feel about it after working with it for a while?